File: mockez-front/src/app/core/model/composite_key/group-member-pk.ts
/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

export class GroupMemberPk {

  private _groupId: string | undefined = undefined;
  private _userId: string | undefined = undefined;

  get groupId(): string | undefined {
    return this._groupId;
  }

  set groupId(value: string | undefined) {
    this._groupId = value;
  }

  get userId(): string | undefined {
    return this._userId;
  }

  set userId(value: string | undefined) {
    this._userId = value;
  }
}

File: mockez-back/src/main/java/com/mocker/domain/model/entity/composite_key/GroupMemberPK.java
package com.mocker.domain.model.entity.composite_key;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.io.Serializable;
import java.util.UUID;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

@Data
@Embeddable
@SuperBuilder
@NoArgsConstructor
public class GroupMemberPK implements Serializable {

    @Column(nullable = false)
    private UUID groupId;

    @Column(nullable = false)
    private UUID userId;
}
